mkosi

```diff
@@ -2640,6 +2667,10 @@ def run_build_script(args, workspace, raw):
         else:
             cmdline.append("--chdir=/root")
 
+        if args.build_dir is not None:
+            cmdline.append("--setenv=BUILDDIR=/root/build")
```

Comment:
well, i wanted that scripts can use the fact whether $BUILDDIR is set or not to decide whether they want to do an in-tree or out-of-tree build, and I think that's really relevant, as doing out-of-tree builds is often broken in projects, as it's "new-style" and some projects aren't up for that yet...

I'd claim that removing mkosi.builddir/ is fine if it has the same effects as removing mkosi.default: i.e. break stuff...